controllers/solrprometheusexporter_controller.go

```diff
@@ -286,6 +301,13 @@ func (r *SolrPrometheusExporterReconciler) SetupWithManagerAndReconciler(mgr ctr
 		return err
 	}
 
+	// Get notified when the basic auth secret updates; exporter pods must be restarted if the basic auth password
+	// changes b/c the credentials are loaded from a Java system property at startup and not watched for changes.
+	ctrlBuilder, err = r.indexAndWatchForBasicAuthSecret(mgr, ctrlBuilder)
```

Comment:
Honestly think at this point we should bake something into ControllerRuntime to automate this. We use it so often.

(Nothing to do with this PR, just an observation)